libqtile/backend/wayland/qw/output.h

```diff
@@ -1,13 +1,25 @@
 #ifndef OUTPUT_H
 #define OUTPUT_H
 
+#include <cairo/cairo.h>
 #include <wayland-server-core.h>
 #include <wlr/backend/headless.h>
+#include <wlr/types/wlr_buffer.h>
 #include <wlr/types/wlr_output.h>
 #include <wlr/types/wlr_scene.h>
 
 struct qw_server;
 
+struct qw_output_background_wallpaper {
+    struct wlr_scene_buffer *buffer;
+    cairo_surface_t *surface;
+};
+
+struct qw_output_background {
+    struct wlr_scene_rect *color_rect;
+    struct qw_output_background_wallpaper *wallpaper;
```

Comment:
Wouldn't it be better for this to be a discriminated union? Can you have both `color_rect` an `wallpaper`?